### PR TITLE
Feature/mpesaToCusd

### PR DESCRIPTION
**Pull request by Alphonce-Mutebi**

### CODE REVIEW

File: App.tsx [hunk 1]
```diff
@@ -77,7 +77,7 @@ export default function App() {
     return (
       <NavigationContainer>
         <Stack.Navigator
-          initialRouteName="Onboarding"
+          initialRouteName="Add Funds"
```
Comment (steffenkrogmann) [inline App.tsx:80]: please change to "Top up".
We are starting to play with wording based on feedback from claire and nirel. not mission critical ;-)
@Alphonce-Mutebi

File: src/components/Cards/Banner.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    const sv = withTiming(0, { duration: 200 });
+  };
+
+  React.useImperativeHandle(ref, () => ({ openBanner, closeBanner }));
+
+  return (
+    <>
+      <Animated.View style={[styles.shadow, AnimatedStyles.shadow]} />
+      <AnimatedLinearGradient
+        colors={["#F7EFFA", "#FCF8ED"]}
```
Comment (steffenkrogmann) [inline src/components/Cards/Banner.tsx:57]: @Alphonce-Mutebi what are these colors? should they be in the colors file?

File: src/components/buttons/KeyPad.tsx [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+        <TouchableHighlight
+          onPress={() => handleDelete()}
+          underlayColor="rgba(247, 239, 250, 1)"
+          delayPressOut={100}
+          style={styles.key}
+        >
+          <Feather
+            name="delete"
+            size={24}
+            color="#1C1939"
```
Comment (steffenkrogmann) [inline src/components/buttons/KeyPad.tsx:133]: @Alphonce-Mutebi what is this color?

File: src/components/buttons/KeyPad.tsx [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+
+  key: {
+    width: keyWidth,
+    height: keyHeight,
+    borderRadius: keyWidth / 2,
+    justifyContent: "center",
+  },
+
+  number: {
+    fontSize: 24,
```
Comment (steffenkrogmann) [inline src/components/buttons/KeyPad.tsx:161]: there is a lot of font / style informateion here from line 161 onward @Alphonce-Mutebi 
should that be imported?

or is the keypad an exception?

is it really?

File: src/screens/mpesaToCusd/AddFunds.tsx [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+};
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    marginHorizontal: 30,
+    justifyContent: "flex-end",
+  },
+
+  title: {
```
Comment (steffenkrogmann) [inline src/screens/mpesaToCusd/AddFunds.tsx:69]: @Alphonce-Mutebi there is also a bunch of styling information from here on onward until the end of the file.

font size
color
font family
why is that? should that better be imported?

File: src/screens/mpesaToCusd/SelectPaymentMethodScreen.tsx [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+    </Fragment>
+  );
+};
+
+const styles = StyleSheet.create({
+  container: {
+    margin: 30,
+  },
+
+  title: {
```
Comment (steffenkrogmann) [inline src/screens/mpesaToCusd/SelectPaymentMethodScreen.tsx:95]: @Alphonce-Mutebi a lot of styling info in here again until the end of the file

File: src/screens/mpesaToCusd/TransactionType.tsx [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+  );
+};
+
+const styles = StyleSheet.create({
+  container: {
+    margin: 30,
+  },
+
+  title: {
+    fontSize: 28,
```
Comment (steffenkrogmann) [inline src/screens/mpesaToCusd/TransactionType.tsx:101]: @Alphonce-Mutebi again styling info here and onward, font size, color, fontfamily, should be imported, rightß

**Review by steffenkrogmann — COMMENTED** [review]
@Alphonce-Mutebi i made a few comments.

It seems to me like there is a bunch of styling information directly coded into the screens files, etc. that should better be imported?

we should centralize things as much as possible.
but when you first want to merge this so that we can get a first build, i leave that up to you.

File: src/screens/mpesaToCusd/SelectPaymentMethodScreen.tsx [hunk 8]
```diff
@@ -0,0 +1,194 @@
...
+  },
+
+  icon: {
+    width: iconWidth,
+    height: iconHeight,
+    borderRadius: 8,
+    marginRight: 15,
+    alignItems: "center",
+    justifyContent: "center",
+    backgroundColor: "#FFFFFF",
```
Comment (Allen-Muhani) [inline src/screens/mpesaToCusd/SelectPaymentMethodScreen.tsx:170]: Get this from the syling files.

File: src/screens/mpesaToCusd/TransactionType.tsx [hunk 9]
```diff
@@ -0,0 +1,164 @@
...
+
+const OperationButton = (props) => {
+  return (
+    <TouchableOpacity onPress={() => props.handleAction()}>
+      <View style={{ flexDirection: "row" }}>
+        <View style={buttonStyles.icon}>
+          <EvilIcons
+            name={props.icon}
+            size={40}
+            color="#4840BB"
```
Comment (Allen-Muhani) [inline src/screens/mpesaToCusd/TransactionType.tsx:19]: Get this from the syling files.

File: src/screens/mpesaToCusd/TransactionType.tsx [hunk 10]
```diff
@@ -0,0 +1,164 @@
...
+    borderWidth: 1,
+    borderRadius: 15,
+    borderColor: "transparent",
+    alignItems: "center",
+  },
+
+  icon: {
+    width: buttonWidth / 2 - 10,
+    height: buttonWidth / 2 - 6,
+    backgroundColor: "#FFFFFF",
```
Comment (Allen-Muhani) [inline src/screens/mpesaToCusd/TransactionType.tsx:140]: Get this from the styling files.

File: src/screens/mpesaToCusd/TransactionType.tsx [hunk 11]
```diff
@@ -0,0 +1,164 @@
...
+    justifyContent: "center",
+    alignItems: "center",
+    borderRadius: 8,
+    marginTop: 15,
+  },
+
+  buttonTitle: {
+    fontSize: 18,
+    lineHeight: 22,
+    color: "#4840BB",
```
Comment (Allen-Muhani) [inline src/screens/mpesaToCusd/TransactionType.tsx:150]: Get this from the styling files.

File: src/screens/mpesaToCusd/TransactionType.tsx [hunk 12]
```diff
@@ -0,0 +1,164 @@
...
+    marginTop: 15,
+    fontFamily: "Rubik_400Regular",
+  },
+
+  buttonSubTitle: {
+    fontSize: 12,
+    lineHeight: 14,
+    color: "#333333",
```
Comment (Allen-Muhani) [inline src/screens/mpesaToCusd/TransactionType.tsx:158]: Get this from the styling files.

File: src/screens/mpesaToCusd/SelectPaymentMethodScreen.tsx [hunk 13]
```diff
@@ -0,0 +1,194 @@
...
+
+  image: {
+    height: 150,
+    maxWidth: SIZES.width * 0.8,
+    resizeMode: "contain",
+    marginBottom: 20,
+  },
+
+  title: {
+    fontSize: 18,
```
Comment (Allen-Muhani) [inline src/screens/mpesaToCusd/SelectPaymentMethodScreen.tsx:121]: Use the title styling in the styling files.

File: src/components/buttons/Swipe.tsx [hunk 14]
```diff
@@ -160,7 +141,7 @@ const styles = StyleSheet.create({
   swipeCont: {
     height: BUTTON_HEIGHT,
     width: BUTTON_WIDTH,
-    backgroundColor: COLORS.white,
+    backgroundColor: "#fff",
```
Comment (Allen-Muhani) [inline src/components/buttons/Swipe.tsx:144]: Use the styling files.

File: src/components/buttons/Swipe.tsx [hunk 15]
```diff
@@ -119,38 +115,23 @@ const SwipeButton = (props) => {
...
       </PanGestureHandler>
       <Animated.Text style={[styles.swipeText, AnimatedStyles.swipeText]}>
-        {props.title}
+        Swipe to confirm
       </Animated.Text>
     </AnimatedLinearGradient>
   );
 };
 
-/**
```
Comment (Allen-Muhani) [inline src/components/buttons/Swipe.tsx:137]: Let this remain. They are for documentation purposes. Someone new using the button does not have to got to the code to check the props being used. They serve as defaults.

File: src/components/buttons/Swipe.tsx [hunk 16]
```diff
@@ -119,38 +115,23 @@ const SwipeButton = (props) => {
...
-const handler = async () => {
-    Alert.alert("Pass handler method on prop (handleAction)")
-}
-
-/**
- * Default values for expected props.
- * @title the title of the button.
- * @handleAction the method to be executed on onEnd call.
- */
-SwipeButton.defaultProps = {
```
Comment (Allen-Muhani) [inline src/components/buttons/Swipe.tsx:149]: Same to this

File: src/components/buttons/Swipe.tsx [hunk 17]
```diff
@@ -119,38 +115,23 @@ const SwipeButton = (props) => {
 
   return (
     <AnimatedLinearGradient
-          style={styles.swipeCont}
-          start={[1, 0]}
-          end={[0, 1]} colors={COLORS.buttonGradient}    >
+      style={styles.swipeCont}
+      colors={["rgba(183, 0, 76, 0.3)", "rgba(19, 63, 219, 1)"]}
```
Comment (Allen-Muhani) [inline src/components/buttons/Swipe.tsx:119]: Let this remain in thje styling sheet for re-usability

File: src/components/Cards/Banner.tsx [hunk 18]
```diff
@@ -0,0 +1,87 @@
...
+    backgroundColor: "#000",
+  },
+
+  modal: {
+    width: "auto",
+    borderRadius: 16,
+    position: "absolute",
+    marginHorizontal: 20,
+    paddingHorizontal: 20,
+    backgroundColor: "#FFF",
```
Comment (Allen-Muhani) [inline src/components/Cards/Banner.tsx:82]: Use colors from the style sheet

**Review by Allen-Muhani — CHANGES_REQUESTED** [review]

**Pull request closed without merging**